exporters/zipkin/src/recordable.cc

```diff
@@ -213,6 +213,16 @@ void Recordable::SetName(nostd::string_view name) noexcept
   span_["name"] = name.data();
 }
 
+void Recordable::SetResource(const opentelemetry::sdk::resource::Resource &resource) noexcept
```

Comment:
Confirming re. thread-safety of `SetResource` vs `GetServiceName` -- both cannot be called at once from two different threads?

Reply:
`SetResource` would be called when Span using `Tracer::StartSpan()`. And GetServiceName() would be when Span is ending : `Span::End() -> Processor::OnEnd() -> Exporter::Export()`. Both these need to be sequential in same thread.

Reply:
I would let it go for now. But we may need to describe some of these expectations where we do not explicitly shield the access with a mutex, esp. in the other place where we return an object (that may get changed in another thread) by reference.

Reply:
Good point. Specifically for Resources, these are immutable as per the specs, with their ownership transferred to TracerProvider while it's initialization. So there won't be race condition arising while accessing them within the pipeline.